gammapy/datasets/simulate.py

```diff
@@ -26,6 +27,66 @@ class MapDatasetEventSampler:
     def __init__(self, random_state="random-seed"):
         self.random_state = get_random_state(random_state)
 
+    def _sample_coord_time_energy(self, dataset, temporal_model, t_delta="1 sec"):
+        energy_axis = dataset.geoms["geom"].axes["energy"]
+
+        time_start, time_stop, time_ref = (dataset.gti.time_start,
+                                           dataset.gti.time_stop,
+                                           dataset.gti.time_ref)
+        t_min = Time(time_start)
+        t_max = Time(time_stop)
+        t_delta = u.Quantity(t_delta)
+
+        ontime = u.Quantity((t_max - t_min).sec, "s")
+
+        time_unit = (
+            u.Unit(self.table.meta["TIMEUNIT"])
+            if hasattr(self, "table")
+            else ontime.unit
+        )
+
+        t_step = t_delta.to_value(time_unit)
+        t_step = (t_step * u.s).to("d")
+
+        t = Time(np.arange(t_min.mjd, t_max.mjd, t_step.value), format="mjd")
+        time_axis = MapAxis.from_edges(t.value * u.d, name="time")
+
+        target = temporal_model.geom.skycoord #change it when the TemplateTemporalModel is updated
+        on_region = PointSkyRegion(center=target)
+        region_geom = RegionGeom.create(on_region,
+                                        axes=[energy_axis, time_axis])
+    
+        flux = temporal_model.evaluate_geom(region_geom)
+        region_exposure = dataset.exposure.to_region_nd_map(region_geom.center_skydir)
+        
+        npred = flux * region_exposure #check the broadcasting
+        data = npred.data[np.isfinite(npred.data)]
+        n_events = self.random_state.poisson(np.sum(data))
+
+        coords = npred.sample_coord(n_events=n_events, random_state=self.random_state)
+
+        position = SkyCoord(coords["lon"], coords["lat"],
```

Comment:
Maybe all the `MapCoord` to table manipulation could be performed utside the function as it is common with the other case